testing-overhead/src/test/java/io/opentelemetry/agents/AgentResolver.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package io.opentelemetry.agents;
+
+import java.io.FileOutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.util.Optional;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+
+public class AgentResolver {
+
+  private final LatestAgentSnapshotResolver snapshotResolver = new LatestAgentSnapshotResolver();
+
+  public Optional<Path> resolve(Agent agent) throws Exception {
+    if(Agent.NONE.equals(agent)){
+      return Optional.empty();
+    }
+    if(Agent.LATEST_SNAPSHOT.equals(agent)){
+      return snapshotResolver.resolve();
+    }
+    if(agent.hasUrl()){
+      return Optional.of(downloadAgent(agent));
+    }
+    throw new IllegalArgumentException("Unknown agent: " + agent);
+  }
+
+  Path downloadAgent(Agent agent) throws Exception {
```

Comment:
Would it make sense to pass just `URL` here? You've already checked that `agent.hasUrl()` and you're only using the URL

Reply:
Sure. It's funny, I had it that way and factored it back. Not sure why. Agree that url makes more sense here, since this method should be private.